modules/generator/processor/localblocks/processor.go

```diff
@@ -238,6 +255,38 @@ func (p *Processor) completeLoop() {
 	}
 }
 
+func (p *Processor) flushLoop() {
+	defer p.wg.Done()
+
+	go func() {
+		<-p.closeCh
+		p.flushqueue.Close()
+	}()
+
+	for {
+		o := p.flushqueue.Dequeue()
+		if o == nil {
+			return
+		}
+
+		op := o.(*flushOp)
+		op.attempts++
+		err := p.flushBlock(op.blockID)
+		if err != nil {
+			_ = level.Error(p.logger).Log("msg", "failed to flush a block", "err", err)
+
+			if op.attempts < 3 {
```

Comment:
Let's copy the behavior of the ingester, which doesn't limit retries on flushing to object storage. This will make it tolerate network or other transient errors as long as needed.

Reply:
Done!

Reply:
Isn't allowing infinite retries dangerous?